python/cudf/cudf/tests/test_parquet.py

```diff
@@ -2588,3 +2590,47 @@ def test_parquet_writer_zstd():
     else:
         got = pd.read_orc(buff)
         assert_eq(expected, got)
+
+
+def test_parquet_writer_time_delta_physical_type():
+    df = cudf.DataFrame(
+        {
+            "s": cudf.Series([1], dtype="timedelta64[s]"),
+            "ms": cudf.Series([2], dtype="timedelta64[ms]"),
+            "us": cudf.Series([3], dtype="timedelta64[us]"),
+            # 4K because Pandas/pyarrow don't support non-zero nanoseconds
+            # in Parquet files
+            "ns": cudf.Series([4000], dtype="timedelta64[ns]"),
+        }
+    )
+    buffer = BytesIO()
+    df.to_parquet(buffer)
+
+    pd.read_parquet(buffer)  # should not throw
```

Comment:
Can we do any kind of assertion (equality, size, etc.) beyond "does not throw"?

Reply:
Added shape and column names checks.
Pandas returns `object` columns here, I don't know how to compare them against hard-coded values.

Reply:
@galipremsagar taught how to compare object Pandas series, added full dataframe check to the PR :)

Reply:
Fantastic!